src/python/pants/backend/python/tasks2/resolve_requirements.py

```diff
@@ -5,153 +5,18 @@
 from __future__ import (absolute_import, division, generators, nested_scopes, print_function,
                         unicode_literals, with_statement)
 
-import os
-import shutil
-
-from pex.fetcher import Fetcher
-from pex.interpreter import PythonInterpreter
-from pex.pex import PEX
-from pex.pex_builder import PEXBuilder
-from pex.platforms import Platform
-from pex.resolver import resolve
-from twitter.common.collections import OrderedSet
-
-from pants.backend.python.python_setup import PythonRepos, PythonSetup
 from pants.backend.python.targets.python_requirement_library import PythonRequirementLibrary
-from pants.base.fingerprint_strategy import (DefaultFingerprintHashingMixin,
-                                             TaskIdentityFingerprintStrategy)
-from pants.invalidation.cache_manager import VersionedTargetSet
-from pants.task.task import Task
-
-
-class PythonRequirementFingerprintStrategy(DefaultFingerprintHashingMixin,
-                                           TaskIdentityFingerprintStrategy):
-
-  def compute_fingerprint(self, req_lib):
-    hash_elements_for_target = []
-    hash_elements_for_target.extend([req.cache_key() for req in req_lib.requirements])
-    if not hash_elements_for_target:
-      return None
-    hasher = self._build_hasher(req_lib)
-    for element in hash_elements_for_target:
-      hasher.update(element)
-    return hasher.hexdigest()
-
-
-class ResolveRequirements(Task):
-  """Resolve 3rd-party Python requirements.
+from pants.backend.python.tasks2.resolve_requirements_task_base import ResolveRequirementsTaskBase
 
-  Creates an (unzipped) PEX on disk containing all the resolved requirements.
-  This PEX can be merged with source PEXes to create a unified Python environment
-  for running the relevant python code.
-  """
 
+class ResolveRequirements(ResolveRequirementsTaskBase):
```

Comment:
Done, although this isn't really intended to be run directly by humans.

We probably need some sort of distinction between goals that are user-facing and goals that aren't.